lib/go/test/service_test.go

```diff
@@ -401,6 +405,99 @@ func TestContracts(t *testing.T) {
 
 	})
 
+	t.Run("Should check if payer has sufficient balance to execute tx", func(t *testing.T) {
+		// create SmallBalanceContract contract
+		code := []byte(`
+			access(all) contract SmallBalanceContract {
+    			access(all) var value: Int32
+
+    			init() {
+        			self.value = 42
+    			}
+
+				access(all) fun SetValue(new_value: Int32) { 
+					self.value = new_value
+				}
+			}
+    	`)
+		keys := test.AccountKeyGenerator()
+		accKey, accSigner := keys.NewWithSigner()
+		accAddress, err := adapter.CreateAccount(context.Background(), []*flow.AccountKey{accKey}, []sdktemplates.Contract{
+			{
+				Name:   "SmallBalanceContract",
+				Source: string(code),
+			},
+		})
+
+		assert.NoError(t, err)
+		_, err = b.CommitBlock()
+		assert.NoError(t, err)
+
+		// we want to execute some tx so that the payer has less balance than default
+		txCode := []byte(fmt.Sprintf(`
+			import SmallBalanceContract from 0x%s
+			
+			transaction(value: Int32) {
+			  prepare(signer: auth(Storage) &Account) {}
+			
+			  execute {
+			   if value > SmallBalanceContract.value {
+				  SmallBalanceContract.SetValue(new_value: value + 1)
+			   } else {
+				  SmallBalanceContract.SetValue(new_value: value - 1)
+			   }
+			  }
+			}
+		`, accAddress))
+
+		tx := flow.NewTransaction().
+			SetScript(txCode).
+			SetGasLimit(9999).
+			SetProposalKey(b.ServiceKey().Address, b.ServiceKey().Index, b.ServiceKey().SequenceNumber).
+			SetPayer(accAddress).
+			AddAuthorizer(accAddress)
+
+		err = tx.AddArgument(cadence.Int32(15))
+		require.NoError(t, err)
+
+		txRes := signAndSubmit(
+			t, b, tx,
+			[]flow.Address{accAddress},
+			[]crypto.Signer{accSigner},
+			false,
+		)
+		require.True(t, txRes.Succeeded())
+
+		// mint some flows to the account
+		//mintTokensForAccount(t, b, env, accAddress, "15.0")
+
+		//TODO: remove this. this is for debug only
+		acc, err := adapter.GetAccount(context.Background(), accAddress)
+		require.NotNil(t, acc)
+		require.NoError(t, err)
+
+		// set up args
+		cadenceAddress := cadence.NewAddress(accAddress)
+		inclusionEffort := cadence.UFix64(20)
```

Comment:
```suggestion
		inclusionEffort := cadence.UFix64(100_000_000) 
```
this should be `1.0 UFix64`